cassiopeia/solver/HybridSolver.py

```diff
@@ -233,52 +281,59 @@ def apply_bottom_solver(
             subproblem_tree.add_edge(root, samples[0])
             return subproblem_tree, root
 
-        subproblem_character_matrix = self.unique_character_matrix.loc[samples]
+        character_matrix = cassiopeia_tree.get_original_character_matrix()
+        subproblem_character_matrix = character_matrix.loc[samples]
 
         subtree_root = data_utilities.get_lca_characters(
             subproblem_character_matrix.loc[samples].values.tolist(),
-            self.missing_char,
+            cassiopeia_tree.missing_state_indicator,
         )
 
-        base_logfile = self.bottom_solver.logfile.split(".log")[0]
+        base_logfile = logfile.split(".log")[0]
         subtree_root_string = "-".join([str(s) for s in subtree_root])
         logfile = f"{base_logfile}_{subtree_root_string}.log"
 
-        subtree_solver = copy.deepcopy(self.bottom_solver)
-        subtree_solver.prepare_for_subproblem(
-            subproblem_character_matrix, logfile
+        subtree = CassiopeiaTree(
+            subproblem_character_matrix,
+            missing_state_indicator=cassiopeia_tree.missing_state_indicator,
+            priors=cassiopeia_tree.priors,
         )
+        self.bottom_solver.solve(subtree)
 
-        subtree_solver.solve()
-
-        subproblem_tree = subtree_solver.tree
+        subproblem_tree = subtree.get_network()
         subproblem_root = [
             n for n in subproblem_tree if subproblem_tree.in_degree(n) == 0
         ][0]
         subproblem_tree.add_edge(root, subproblem_root)
 
         return subproblem_tree, root
 
-    def assess_cutoff(self, samples: List[str]) -> bool:
+    def assess_cutoff(
```

Comment:
In general, which methods are public and which are private? I think it is important that we make this clear.

Reply:
I think anything not preceded by `__` will be public.